src/Externals/spire/arc-look-at/ArcLookAt.cpp

```diff
@@ -103,10 +104,12 @@ void ArcLookAt::doZoom(const glm::vec2& ssPos)
 //------------------------------------------------------------------------------
 glm::mat4 ArcLookAt::getWorldViewTransform() const
 {
-  glm::mat4 finalTrafo;
-  finalTrafo[3].xyz() = -mCamLookAt;
-  finalTrafo          = mArcBall->getTransformation() * finalTrafo;
-  finalTrafo[3][2]   += -mCamDistance;
+  glm::mat4 finalTrafo(1.0);
+  for (int i = 0; i < mCamLookAt.length(); ++i)
+    finalTrafo[3][i] = -mCamLookAt[i];
```

Comment:
I'm trying to get better with this magic number thing. However, I notice here on line 109 and line 112, the index [3] is used. Does this have some sort of special significance? And if it does can this number be given a name?

Reply:
It seems in the graphics code 0/1/2/3 mapping to x/y/z/w coordinates is common, although there's no reason to use the raw integers everywhere. Just define a top-level enum so it looks like `finalTrafo[Coordinate.w][i] = ...` although that's a bit wordy

Reply:
ah, I see. yeah, even a `Coord.w` would be a lot more clear. But if it isn't necessary right now then I wouldn't worry about it too much.

Reply:
The x/y/z/w convention is a bit strange for a homogeneous transformation. The literature typically uses numbers for indexing. In this case, finalTrafo[3] is the translational component of the transformation. We could do Coord.x/y/z/trans, but the bottom row is a special case so that convention could get confusing as well.